storage/range.go

```diff
@@ -1548,3 +1593,109 @@ func (r *Range) SetHardState(st raftpb.HardState) error {
 	return engine.MVCCPutProto(r.rm.Engine(), nil, engine.RaftStateKey(r.Desc.RaftID),
 		proto.ZeroTimestamp, nil, &st)
 }
+
+// intersectReplicaSets is used in AdminMerge to ensure that the ranges are
+// all colocated on the same set of replicas.
+func intersectReplicaSets(a, b []proto.Replica) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	set := make(map[int32]int)
+	for _, replica := range a {
+		set[replica.StoreID]++
+	}
+
+	for _, replica := range b {
+		set[replica.StoreID]--
+	}
+
+	for _, value := range set {
+		if value != 0 {
+			return false
+		}
+	}
+
+	return true
+}
+
+// AdminMerge extends the range to subsume the range that comes next in
+// the key space.  The range being subsumed is provided in args.SubsumedRange.
```

Comment:
I always put two spaces after a period, especially in a monospaced font.  But I can go through and adjust them all if you want.